js/coinsbit.js

```diff
@@ -0,0 +1,1103 @@
+            //              market: 'ETH_USDT',
+            //              makerFee: '0.002',
+            //              dealFee: '0.01597712',
+            //              dealStock: '0.002',
+            //              dealMoney: '7.98856357',
+            //              marketName: 'ETH_USDT'
+            //          },
+            //          ...
+            //        ]
+            //     },
+            //     code: '200',
+            // }
+            const data = this.safeValue (response, 'result', {});
+            const records = this.safeValue (data, 'records', []);
+            return this.parseOrders (records, undefined, since, limit, params);
+        }
+    }
+
+    parseOrder (order, market = undefined) {
+        // createOrder & cancelOrder  ( & fetchOpenOrders, but 'id' instead of 'orderId')
+        //     {
+        //       orderId: '8629354782',
+        //       market: 'ETH_USDT',
+        //       price: '4000',
+        //       side: 'buy',
+        //       type: 'limit',
+        //       timestamp: '1640212275.928',
+        //       dealMoney: '7.98856357',
+        //       dealStock: '0.002',
+        //       amount: '0.002',
+        //       takerFee: '0.002',
+        //       makerFee: '0.002',
+        //       left: '0',
+        //       dealFee: '0'
+        //     }
+        //
+        //  fetchClosedOrder
+        //      {
+        //         dealOrderId: '8631729321', //<--- this is different than passed order-id
+        //         id: '1595921',
+        //         time: '1640258718.402',
+        //         fee: '0.01572657',
+        //         price: '3931.64271609',
+        //         amount: '0.002',
+        //         role: '2',
+        //         deal: '7.86328543'
+        //      }
+        //
+        //  fetchClosedOrders
+        //      {
+        //         amount: '0.002',
+        //         price: '4000',
+        //         type: 'limit',
+        //         id: '8629354787',
+        //         side: 'buy',
+        //         ctime: '1640212275.928',
+        //         takerFee: '0.002',
+        //         ftime: '1640212275.93',
+        //         market: 'ETH_USDT',
+        //         makerFee: '0.002',
+        //         dealFee: '0.01597712',
+        //         dealStock: '0.002',
+        //         dealMoney: '7.98856357',
+        //         marketName: 'ETH_USDT'
+        //      },
+        const isFromFCOs = ('ctime' in order);
+        let id = undefined;
+        if (isFromFCOs) {
+            id = this.safeString (order, 'id');
+        } else {
+            id = this.safeString (order, 'orderId');
+        }
+        const side = this.safeString (order, 'side');
+        const orderType = this.safeString (order, 'type');
+        let timestamp = this.safeTimestamp2 (order, 'timestamp', 'time');
+        if (timestamp === undefined) {
+            timestamp = this.safeTimestamp2 (order, 'ctime', 'ftime');
```

Comment:
Most often `ftime` means filling time, so it should go to `lastTradeTimestamp`, and it shouldn't go to `timestamp`, cause the `timestamp` field should always contain the order placement timestamp.